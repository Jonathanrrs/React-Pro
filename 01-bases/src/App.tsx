

function App() {
  return (
    <>
    <h1>React tsx</h1>
    <hr />
    </>
  );
}

export default App;
