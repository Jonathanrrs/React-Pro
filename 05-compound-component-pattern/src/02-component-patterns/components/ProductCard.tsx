import { ReactElement } from 'react'; /* es una interfaz de tipo generico */
import { useProduct } from '../hooks/useProduct';

import styles from '../styles/styles.module.css';
import noImage from '../assets/no-image.jpg';

interface Props {
    product: Product;
    children?: ReactElement | ReactElement[]
}

interface Product {
    id: string;
    title: string,
    img?: string
}

/* '' asi para que la img sea opcional, vacio para el ternario es que no tiene valor */
export const ProductImage = ({ img = '' }) => {
    return (
        <img className={styles.productImg} src={img ? img : noImage} alt="Product image" />
    );
}

/* se le obliga que venga el titulo, es una interfaz */
export const ProductTitle = ({ title }: { title: string }) => {
    return (
        <span className={styles.productDesciption}>{title}</span>
    );
}

interface ProductButtonsProps {
    counter: number;
    /* funcion que no regresa nada */
    increaseBy: (value: number) => void;
}

export const ProductButtons = ({counter, increaseBy}: ProductButtonsProps) => {
    return (
        <div className={styles.buttonsContainer}>
            <button
                className={styles.buttonMinus}
                onClick={() => increaseBy(-1)}
            >
                -</button>
            <div className={styles.countLabel}>{counter}</div>

            <button
                className={styles.buttonAdd}
                onClick={() => increaseBy(+1)}
            >
                +</button>
        </div>
    );
}

export const ProductCard = ({ children, product }: Props) => {

    const { counter, increaseBy } = useProduct();

    return (
        <div className={styles.productCard}>
            {children}

            {/* <ProductImage img={product.img} />
            <ProductTitle title={product.title} />

            <ProductButtons counter={counter} increaseBy={increaseBy} /> */}

        </div>
    )
}
