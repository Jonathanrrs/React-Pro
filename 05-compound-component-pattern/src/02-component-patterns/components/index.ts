export { ProductTitle } from './ProductTitle';
export { ProductButtons } from './ProductButtons';
export { ProductCard } from './ProductCard';
export { ProductImage } from './ProductImage';

