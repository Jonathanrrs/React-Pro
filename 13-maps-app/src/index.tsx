import React from 'react';
import ReactDOM from 'react-dom';
import { MapsApp } from './MapsApp';

import mapboxgl from 'mapbox-gl'; 
 
mapboxgl.accessToken = `${process.env.REACT_APP_TOKEN_MAPS}`;

if(!navigator.geolocation) {
  alert('Tu navegador no tiene opción de Geolocation');
  throw new Error('Tu navegador no tiene opción de Geolocation')
}

ReactDOM.render(
  <React.StrictMode>
    <MapsApp />
  </React.StrictMode>,
  document.getElementById('root')
);


