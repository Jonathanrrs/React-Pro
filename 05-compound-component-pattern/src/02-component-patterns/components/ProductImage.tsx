import { useContext } from "react";
import { ProductContext } from "./ProductCard";

import noImage from '../assets/no-image.jpg';
import styles from '../styles/styles.module.css';

/* '' asi para que la img sea opcional, vacio para el ternario es que no tiene valor */
export const ProductImage = ({ img = '' }) => {

    const {product} = useContext(ProductContext);
    let imgToShow: string;

    if(img) {
        imgToShow = img;
    } else if(product.img) {
        imgToShow = product.img
    } else {
        imgToShow = noImage;
    }


    return (
        <img className={styles.productImg} src={imgToShow} alt="Product image" />
    );
}