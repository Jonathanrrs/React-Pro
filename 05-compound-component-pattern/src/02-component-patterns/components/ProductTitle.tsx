import { useContext } from "react";
import { ProductContext } from "./ProductCard";

import styles from '../styles/styles.module.css';

/* es una interfaz */
export const ProductTitle = ({ title }: { title?: string }) => {

    const {product} = useContext(ProductContext);

    return (
        <span className={styles.productDesciption}>{title ? title : product.title}</span>
    );
}