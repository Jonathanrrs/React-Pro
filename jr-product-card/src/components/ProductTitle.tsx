import React, { useContext, CSSProperties } from "react";
import { ProductContext } from "./ProductCard";

import styles from '../styles/styles.module.css';

export interface Props {
    title?: string;
    className?: string;
    style?: CSSProperties;
}

/* es una interfaz */
export const ProductTitle = ({ title, className, style }: Props) => {

    const { product } = useContext(ProductContext);

    return (
        <span
            className={`${styles.productDesciption} ${className}`}
            style={style}
        >{title ? title : product.title}
        </span>
    );
}