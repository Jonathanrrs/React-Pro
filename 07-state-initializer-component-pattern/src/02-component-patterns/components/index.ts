import { ProductCard as ProductCardHOC } from './ProductCard';
import { ProductCardHOCProps } from '../interfaces/interfaces';

import { ProductTitle } from './ProductTitle';
import { ProductButtons } from './ProductButtons';
import { ProductImage } from './ProductImage';


export { ProductTitle } from './ProductTitle';
export { ProductButtons } from './ProductButtons';
export { ProductImage } from './ProductImage';

/* asignarle nuevas propiedades asi de esta manera */
export const ProductCard: ProductCardHOCProps = Object.assign(ProductCardHOC, {
    /* capitalizado porque son componentes */
    Title: ProductTitle,
    Image: ProductImage,
    Buttons: ProductButtons
});

/* solo otra manera de exportarlo */
export default ProductCard;

