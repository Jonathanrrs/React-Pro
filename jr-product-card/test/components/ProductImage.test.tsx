import * as React from 'react';
import renderer from 'react-test-renderer';
import { ProductImage, ProductCard } from '../../src/components';
import { product2 } from '../data/products';


describe('Pruebas en ProductImage', () => {
   
    test('Debe mostrar el componente correctactemente con la imagen personalizada', () => {
        
        const wrapper = renderer.create(
            <ProductImage img="https://hola.jpg" />
        );
        
        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    test('Debe mostrar el component con la imagen del producto ', () => {
       
        const wrapper = renderer.create(
            <ProductCard product={product2}>
                {
                    () => (
                        <ProductImage />
                    )
                } 
            </ProductCard>
        );
        
        expect(wrapper.toJSON()).toMatchSnapshot();
        
    });
    
});
