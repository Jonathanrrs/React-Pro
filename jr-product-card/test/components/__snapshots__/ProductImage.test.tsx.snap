// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en ProductImage Debe mostrar el component con el nombre del producto  1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product image"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`Pruebas en ProductImage Debe mostrar el component con la imagen del producto  1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product image"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`Pruebas en ProductImage Debe mostrar el componente correctactemente con la imagen 1`] = `
<img
  alt="Product image"
  className="productImg undefined"
  src="https://hola.jpg"
/>
`;

exports[`Pruebas en ProductImage Debe mostrar el componente correctactemente con la imagen personalizada 1`] = `
<img
  alt="Product image"
  className="productImg undefined"
  src="https://hola.jpg"
/>
`;
